ui/js/controllers/bugfiler.js

```diff
@@ -0,0 +1,224 @@
+"use strict";
+
+treeherder.controller('BugFilerCtrl', [
+    '$scope', '$rootScope', '$uibModalInstance', '$http', 'summary', 'thBugzillaProductObject',
+    'thPinboard', 'thEvents', 'fullLog', 'parsedLog', 'reftest', 'selectedJob', 'allFailures',
+    'thNotify', 'ThLog',
+    function BugFilerCtrl(
+        $scope, $rootScope, $uibModalInstance, $http, summary, thBugzillaProductObject,
+        thPinboard, thEvents, fullLog, parsedLog, reftest, selectedJob, allFailures,
+        thNotify, ThLog) {
+
+        var $log = new ThLog("BugFilerCtrl");
+
+        /**
+         *  'enter' from the product search input should initiate the search
+         */
+        $scope.productSearchEnter = function(ev) {
+            if (ev.keyCode === 13) {
+                $scope.findProduct();
+            }
+        };
+
+        /*
+         **
+         */
+        $scope.isReftest = function() {
+            return reftest !== "";
+        };
+
+        /**
+         *  Pre-fill the form with information/metadata from the failure
+         */
+        $scope.initiate = function() {
+            $uibModalInstance.parsedSummary = $uibModalInstance.parseSummary(summary);
+
+            $scope.modalSummary = "Intermittent " + $uibModalInstance.parsedSummary[0].join(" | ");
+
+            $("#modalParsedLog").next().attr("href", parsedLog);
+            $("#modalFullLog").next().attr("href", fullLog);
+            if ($scope.isReftest()) {
+                $("#modalReftestLog").next().attr("href", reftest);
+            }
+
+            var thisFailure = "";
+            for(var i = 0; i < allFailures.length; i++) {
+                var omittedLeads = ["TEST-UNEXPECTED-FAIL", "PROCESS-CRASH", "TEST-UNEXPECTED-ERROR", "TEST-UNEXPECTED-TIMEOUT"];
```

Comment:
if ``omittedLeads`` is declared at the top level, you don't have to re-specify it later in the ``parseSummary`` function.